using i2c standard frequency as the default frequency of i2c bus

yassine-safraoui opened the pull request:
Fixes zephyrproject-rtos/zephyr#99216.

This PR adds the default I2C bus clock frequency to the XMC47 relax kit DTS so that when developers enable the I2C peripheral, it automatically uses the standard frequency without requiring them to set it manually. It’s a small, beginner‑friendly change that improves usability and ensures consistent I2C behavior out of the box.

@ljd42 Feel free to give feedback on this if this is not the change you were expecting.

ljd42 commented:
@yassine-safraoui: First of all, many thanks for your  PR! 
They are 4 failing  checks that need to be corrected first, before we consider reviewing your PR:
1.  submit against main, not 4.2. 
2.  please follow the compliance rules for commit message. I suggest you read the [Contributions Guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html), in particular from [Pull Requests and Issues](https://docs.zephyrproject.org/latest/contribute/guidelines.html#pull-requests-and-issues) to [Running CI tests locally](https://docs.zephyrproject.org/latest/contribute/guidelines.html#running-ci-tests-locally). The `compliance.py` script is very handy to check locally  that your PR looks good from compliance point of view.
3. squash your commits.  
4. I challenge the proposed resolution. See my review comments. 
5. Some twister tests are failings. Let's first address the previous points 1) to 4), then we'll see. 

If you have (understanding) questions, reach out. 

Thanks in advance.

ljd42 commented on boards/infineon/xmc47_relax_kit/xmc47_relax_kit.dts:
There's no need to change board dts file. Looking at XMC4 I2C driver here:
https://github.com/zephyrproject-rtos/zephyr/blob/af04b317e9a7054fec378efd0aa4079ed42d2880/drivers/i2c/i2c_ifx_xmc4.c#L473

You see that if the `clock-frequency` property is not explicitly set in the devicetree, the default value is `I2C_SPEED_STANDARD`. So where's the bug? 

**Hint**: check where `I2C_SPEED_STANDARD` is defined / what is its numerical value, and check in `i2c_ifx_xmc4.c` what macro should be used instead.

yassine-safraoui commented:
Hi @ljd42,

- I’ve read the contribution guidelines and updated the commit message to follow them  
- I’ve squashed my commits  
- I’ve changed the PR base to `main`
- I changed the PR's description to link against existing issue
- I’ve proposed an alternative fix that I believe matches what you were suggesting  

Please let me know if there’s anything else you’d like me to do.  
Thanks again for your guidance on my first PR, I appreciate it!

yassine-safraoui committed:
drivers: i2c: xmc47_relax_kit: fix clock frequency macro

The xmc47_relax_kit i2c driver is incorrectly using the
I2C_SPEED_STANDARD macro to set the clock frequency. The correct macro
to use is XMC4_I2C_SPEED_STANDARD which is defined by the XMC4 driver.

Signed-off-by: Yassine Safraoui <yassine.safraoui@grenoble-inp.org>

ljd42 commented:
@yassine-safraoui : LGTM. There's seem to be an (unrelated) issue with the workflow. Let me double-check on Discord pr-help.
